src/vs/workbench/contrib/search/browser/searchModel.ts

```diff
@@ -2100,7 +2101,7 @@ export class SearchModel extends Disposable {
 
 		const start = Date.now();
 
-		Promise.race([asyncResults, Event.toPromise(progressEmitter.event)]).finally(() => {
+		Promise.race([asyncResults, new Promise(resolve => this._register(Event.once(progressEmitter.event)(resolve)))]).finally(() => {
```

Comment:
Same lifetime question in this function. Maybe you want to make a `DisposableStore` for the stuff used internally in the function and dispose when it returns?